opensrp-app/src/test/java/org/smartregister/service/HTTPAgentTest.java

```diff
@@ -560,4 +566,342 @@ public void testOauth2authenticateRefreshTokenInvokesOauth2authenticateCoreWithC
+
+        Assert.assertNotNull(loginResponse.payload().time);
+        Assert.assertEquals("2020-06-02 08:21:40", loginResponse.payload().time.getTime());
+        Assert.assertEquals("Africa/Nairobi", loginResponse.payload().time.getTimeZone());
+
+        Assert.assertNotNull(loginResponse.payload().locations);
+        Assert.assertNotNull(loginResponse.payload().locations.getLocationsHierarchy());
+
+        Assert.assertNotNull(loginResponse.payload().jurisdictions);
+        Assert.assertEquals(1, loginResponse.payload().jurisdictions.size());
+        Assert.assertNotNull("Health Team Kasarani", loginResponse.payload().jurisdictions.get(0));
+
+        Assert.assertNotNull(loginResponse.payload().team);
+        Assert.assertEquals("93c6526-6667-3333-a611112-f3b309999999", loginResponse.payload().team.identifier);
+        Assert.assertEquals("93c6526-6667-3333-a611112-f3b309999999", loginResponse.payload().team.uuid);
+
+        Assert.assertEquals("SUCCESS", loginResponse.name());
+
+        ArgumentCaptor<String> headerKey = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<String> headerValue = ArgumentCaptor.forClass(String.class);
+
+        Mockito.verify(httpsURLConnection).setRequestProperty(headerKey.capture(), headerValue.capture());
+        String capturedKey = headerKey.getValue();
+        String capturedValue = headerValue.getValue();
+
+        Assert.assertEquals(capturedKey, AllConstants.HTTP_REQUEST_HEADERS.AUTHORIZATION);
+        Assert.assertEquals(capturedValue, AllConstants.HTTP_REQUEST_AUTH_TOKEN_TYPE.BEARER + " " + SAMPLE_TEST_TOKEN);
+
+    }
+
+    @Test
+    public void testFetchUserDetailsConstructsCorrectResponseForUnauthorizedRequests() throws Exception {
+
+        URL url = PowerMockito.mock(URL.class);
+        Assert.assertNotNull(url);
+
+        HTTPAgent httpAgentSpy = Mockito.spy(httpAgent);
+
+        Mockito.doReturn(httpsURLConnection).when(httpAgentSpy).getHttpURLConnection(USER_DETAILS_ENDPOINT);
+        Mockito.doReturn(HttpURLConnection.HTTP_UNAUTHORIZED).when(httpsURLConnection).getResponseCode();
+
+        LoginResponse loginResponse = httpAgentSpy.fetchUserDetails(USER_DETAILS_ENDPOINT, SAMPLE_TEST_TOKEN);
+
+        Assert.assertNotNull(loginResponse);
+        Assert.assertNotNull(loginResponse.message());
+        Assert.assertEquals("Please check the credentials", loginResponse.message());
+        Assert.assertNull(loginResponse.payload());
+
+        Assert.assertEquals("UNAUTHORIZED", loginResponse.name());
+
+    }
+
+    @Test
+    public void testFetchUserDetailsConstructsCorrectResponseForRandomServerError() throws Exception {
+
+        URL url = PowerMockito.mock(URL.class);
+        Assert.assertNotNull(url);
+
+        HTTPAgent httpAgentSpy = Mockito.spy(httpAgent);
+
+        Mockito.doReturn(httpsURLConnection).when(httpAgentSpy).getHttpURLConnection(USER_DETAILS_ENDPOINT);
+
+        Mockito.doReturn(errorStream).when(httpsURLConnection).getInputStream();
```

Comment:
I think this should get `getErrorStream`

I also think that the loginresponse for an internal error is a custom message & not the message checked in this test. Kindly fix and confirm if the assumptions here are correct

Reply:
This might be a duplicate with test on line 880

Reply:
Fixed the first one

880 tests a different method with has a similar response flow